source/packaging-software.adoc

```diff
@@ -981,7 +981,7 @@ accessed.
 
 NOTE: You will notice below that we are hard coding the library path. There are
 various methods to avoid needing to do this, many of which are addressed in
-xref:advanced-topics[], under the xref:more-on-macros[] section, and are
+<<advanced-topics>>, under the xref:more-on-macros[] section, and are
```

Comment:
@jeffesquivels 
Hello Jeffrey,

What is the reason for this change?

Reply:
Hi @Daskyleia, that change is meant to correct this issue:

![screenshot from 2018-03-23 13-56-05](https://user-images.githubusercontent.com/4045494/37850793-2834986e-2ea2-11e8-98d2-a4a388e54f6e.png)

Which you can see "live" here: https://rpm-packaging-guide.github.io/#working-with-spec-files

It seems as if the pattern matching algorithm for what should go into the square brackets of `xref:advanced-topics[]` gets confused when they're empty, and mistakenly assumes `], under the xref:more-on-macros[` is the actual "content" inside `xref:advanced-topics`'s square brackets.

This may be a bug in the parser, in which case the best option would be to fix the bug, but a quick search on Google didn't turned out any useful information to do this quickly, so I chose to implement this workaround in the meantime :).

Reply:
Hello @jeffesquivels 

thank you for your answer. I see that the syntax indeed does not work.

Reply:
Hi @Daskyleia 

You're welcome! If there is a better way to fix this, please let me know and I'll take care of it.

Regards,